nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutSyslog.java

```diff
@@ -150,6 +149,7 @@
                     "messages will be sent over a secure connection.")
             .required(false)
             .identifiesControllerService(SSLContextService.class)
+            .dependsOn(PROTOCOL, TCP_VALUE)
```

Comment:
I noticed that in ListenSyslog, the SSL Context Service property is always available, but in PutSyslog it's conditional on TCP being set.  Can you add the condition to both?

Also, if you have an SSL Context Service selected and then change it back to UDP, the processor is marked as invalid even though the SSL Context Service property is no longer available to be configured.  Is it possible to unset this property if protocol is set to UDP, or is that just a limitation of the current "dependsOn" functionality?

Reply:
Good catch, I had intended to update both processors to make SSL Context Service depend on TCP.

I noticed the issue with SSL Context Service and changing from TCP to UDP.  It seems like that is something that might make sense to address at the framework level.